### ISSUE TITLE
JSON data type represented as string in GraphQL

### ISSUE THREAD

**Issue by Hades32** [edgedb/edgedb#2782]
When I have defined a `property data -> json` I would expect this data to be returned as unescaped JSON.

Actual behavior: It's returned as a string:
```
{
  "data": {
    "Person": [
      {
        "name": "Zendaya",
        "data": "{\"random\": 43}"
      }
    ]
  }
}
```
This is in contrast to the json mode of the cli:
```
./cli.sh -j -c 'select Person{name,data} filter .name="Zendaya";'
[{"data": {"random": 43}, "name": "Zendaya"}]
```

CLI version: edgedb-cli 1.0.0-beta.2+d20210725.g047f61ab0
EdgeDB version: 1.0-beta.3+dev5810.d2021072500.g5c1408a6c.cv202107231400

**Comment by elprans** [comment]
There's been a discussion about this [earlier](https://github.com/edgedb/edgedb/issues/592#issuecomment-481330520), we need to revisit this and reconsider using custom JSON scalars in GraphQL.

**Comment by elprans** [comment]
Fixed in #2790!  JSON handling in GraphQL is now consistent with EdgeQL JSON mode.
